test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/HtmlHelperInputExtensionsTest.cs

```diff
@@ -0,0 +1,324 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.AspNet.Mvc.TestCommon;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core
+{
+    /// <summary>
+    /// Test the <see cref="HtmlHelperInputExtensions" /> class.
+    /// </summary>
+    public class HtmlHelperInputExtensionsTest
+    {
+        [Fact]
+        public void CheckboxHelpers_UsesSpecifiedExpression()
+        {
+            // Arrange
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper();
+
+            // Act
+            var checkboxResult = helper.CheckBox("Property1");
+
+            // Assert
+            Assert.Equal(
+                "<input id=\"HtmlEncode[[Property1]]\" name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[checkbox]]\" value=\"HtmlEncode[[true]]\" />" +
+                "<input name=\"HtmlEncode[[Property1]]\" type=\"HtmlEncode[[hidden]]\" value=\"HtmlEncode[[false]]\" />",
+                HtmlContentUtilities.HtmlContentToString(checkboxResult));
```

Comment:
Not a great approach for testing extension methods. As we discussed offline, we have some tests in this folder following the pattern you're using here. But it's better to clearly separate tests of the actual `IHtmlHelper` and `IHtmlHelper<T>` methods from tests of the extensions.

To do that, suggest you either mock `IHtmlGenerator` (see `HtmlHelperFormExtensionsTest`) or `IHtmlHelper` / `IHtmlHelper<T>` (see `HtmlHelperPartialExtensionsTest`). I prefer the second because it's more straightforward.